Code/Yahoo.Yui.Compressor.Tests/JavaScriptCompressorTaskTests.cs

```diff
@@ -182,6 +182,23 @@ public void An_Error_Is_Not_Logged_If_The_Output_File_Has_The_Same_Name_But_A_Di
             Assert.That(BuildEngine.DoesNotContainError(compressor.BuildEngine, ("Output file cannot be the same as source file(s).")));
         }
 
+        [Test]
+        public void If_OutputFile_Directory_Does_Not_Exist_It_Is_Created()
+        {
+            // Arange
+            var compressor = CreateCompressorTask();
+            compressor.SourceFiles = new ITaskItem[] { new TaskItem("Javascript Files\\SampleJavaScript1.js") };
+            compressor.OutputFile = "NewOutputDir\\Output.js";
+            if (Directory.Exists(Path.GetFullPath("NewOutputDir")))
+                Directory.Delete(Path.GetFullPath("NewOutputDir"), true);
+
+            // Act
+            compressor.Execute();
+
+            // Assert
+            Assert.That(BuildEngine.ContainsMessage(compressor.BuildEngine, ("Creating missing directory")));

```

Comment:
The presence of the message is kind of irrelevant - shouldn't you be asserting that the directory was actually created?  That is after all the stated intent in the test name - that the directory is created if it does not exist, but you do not test this.
The fact that you wrote out a message saying the folder was created does not mean the folder was actually created.